.github/actions/e2e_mini/action.yml

```diff
@@ -0,0 +1,65 @@
+name: E2E QEMU test
+description: "Test MiniConstellation functionality on Azure VM."
+
+inputs:
+  azureClientID:
+    description: "Azure client to use for login with OIDC"
+    required: true
+  azureSubscriptionID:
+    description: "Azure subscription to use for login with OIDC"
+    required: true
+  azureTenantID:
+    description: "Azure tenant to use for login with OIDC"
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install terraform
+      uses: hashicorp/setup-terraform@633666f66e0061ca3b725c73b2ec20cd13a8fdd1 # v2.0.3
+      with:
+        terraform_wrapper: false
+
+    - name: Setup Go environment
+      uses: actions/setup-go@c4a742cab115ed795e34d4513e2cf7d472deb55f # tag=v3.3.1
+      with:
+        go-version: "1.19.3"
+
+    - name: Build Constellation CLI
+      uses: ./.github/actions/build_cli
+
+    - name: MiniConstellation E2E
+      shell: bash
+      run: |
+        echo "::group::Terraform"
+        cd .github/actions/e2e_mini
+        # Create SSH key which is uploaded to Azure VM
+        ssh-keygen -t rsa -f id_rsa -N ""
+        terraform init
+        terraform apply -auto-approve
+        export AZURE_VM_IP=$(terraform output -raw public_ip)
+        echo "::endgroup::"
+
+        echo "::group::Run E2E Test"
+        ./wait-for-ssh.sh
+        # Copy locally build Constellation CLI and run e2e script.
+        scp -i id_rsa ../../../build/constellation adminuser@${AZURE_VM_IP}:.
+        ssh -i id_rsa adminuser@${AZURE_VM_IP} 'bash -s' < run-e2e.sh
+        echo "::endgroup::"
+      env:
+        ARM_CLIENT_ID: ${{ inputs.azureClientID }}
+        ARM_SUBSCRIPTION_ID: ${{ inputs.azureSubscriptionID }}
+        ARM_TENANT_ID: ${{ inputs.azureTenantID }}
+
+    - name: Delete deployment
+      if: always()
```

Comment:
Did you check that this works? Afaik the `if: always` only works as expected in workflows...

Reply:
I didn't see any dangling resources in Azure, and [failing pipelines cleaned up after themselves](https://github.com/edgelesssys/constellation/actions/runs/3628778565/jobs/6120193647).